tests/aziot/aziotenroll.go

```diff
@@ -0,0 +1,257 @@
+	// Parse the connection string
+	hostName, sharedAccessKeyName, sharedAccessKey, err := parseConnectionString(connectionString)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse connection string: %v", err)
+	}
+
+	expiry := (time.Now().Unix() + 3600) * int64(hours)
+	sasToken, err := generateSasToken(hostName, sharedAccessKeyName, sharedAccessKey, expiry)
+	if err != nil {
+		return "", fmt.Errorf("failed to generate SAS token: %v", err)
+	}
+
+	return sasToken, nil
+}
+
+// addTPMEnrollment adds a new TPM enrollment to Azure IoT DPS.
+func addTPMEnrollment(enrollmentID, endorsementKey, provService, sasToken string) error {
+	url := fmt.Sprintf(enrollmentAPIURL, provService, enrollmentID)
+
+	// Prepare the enrollment body for TPM attestation
+	enrollment := map[string]interface{}{
+		"registrationId": enrollmentID,
+		"attestation": map[string]interface{}{
+			"type": "tpm",
+			"tpm": map[string]interface{}{
+				"endorsementKey": endorsementKey,
+			},
+		},
+		"provisioningStatus": "enabled",
+		"allocationPolicy":   "hashed",
+		"capabilities": map[string]bool{
+			"iotEdge": false,
+		},
+		"reprovisionPolicy": map[string]bool{
+			"migrateDeviceData":   true,
+			"updateHubAssignment": true,
+		},
+	}
+
+	enrollmentJSON, err := json.Marshal(enrollment)
+	if err != nil {
+		return fmt.Errorf("failed to marshal enrollment: %v", err)
+	}
+
+	req, err := http.NewRequest("PUT", url, bytes.NewBuffer(enrollmentJSON))
+	if err != nil {
+		return fmt.Errorf("failed to create HTTP request: %v", err)
+	}
+	req.Header.Set("Authorization", sasToken)
+	req.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to send HTTP request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusCreated {
+		body, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("failed to create enrollment: %v", string(body))
+	}
+
+	return nil
+}
+
+// deleteEnrollment deletes an enrollment from Azure IoT DPS.
+func deleteEnrollment(enrollmentID, provService, sasToken string) error {
+	url := fmt.Sprintf(enrollmentAPIURL, provService, enrollmentID)
+
+	req, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create HTTP request: %v", err)
+	}
+	req.Header.Set("Authorization", sasToken)
+	req.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to send HTTP request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNoContent {
+		body, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("failed to delete enrollment: %v", string(body))
+	}
+
+	return nil
+}
+
+func readPublicKey(handle tpmutil.Handle) ([]byte, error) {
+	// unfortunaly we can't used SWTPM socket directly, it is blocked becuse
```

Comment:
Minor yetus changes

Reply:
will fix all/squash after the review is completed.